### ISSUE TITLE
Ability to use modded fluid tank as fuel

### ISSUE THREAD

**Issue by Bob74** [Luke100000/ImmersiveAircraft#95]
It would be amazing to be able to use a portable fluid tank to refill the engines like the bucket does. For example, putting a 10.000mB tank in the furnace slot would give back a 9.000mB tank (maybe keep it in the slot and reuse it when needed?).

At the moment we need to put the tank on the ground, use a bucket and fill the engine so it cannot be done while flying 🙂

(And thank you very much for this incredible mod!)

**Comment by krokots** [comment]
Maybe you should have a different engine type that accepts fluids as fuel.

**Comment by Luke100000** [comment]
Since such items store fuel etc in nbt tags, it should be possible to add this as a special fuel, similar how buckets already get a bit of an extra treatment. Please send me a list of tanks and their mods which I should support.

I will start with https://github.com/Dremoline/PortableTanks/blob/1.16/src/main/java/com/dremoline/portabletanks/PortableTankItem.java

**Comment by Bob74** [comment]
I personally use Lightman Currency Tech to add economy to the server and its portable tanks are useful to move and sell fluids.
https://github.com/Lightman314/LCTech/blob/main/src/main/java/io/github/lightman314/lctech/common/blocks/FluidTankBlock.java

They are available in 3 blocks:
- lctech:iron_tank
- lctech:gold_tank
- lctech:diamond_tank

I've just tested in game, we can transfert fluids easily from PortableTanks to LCTech (right-clicking with a tank containing fluids is enough to transfert from one to another)

If it forces you to maintain multiple mods compatibility, you should focus on PortableTanks since players will be able to fill them using other modded tanks.


Thank you for looking into this btw!

**Comment by SammieJay** [comment]
Having support for fluid tanks would be so sick. I've actually been trying to figure out a way to use fuels from [Create: Diesel Generators](https://www.curseforge.com/minecraft/mc-mods/create-diesel-generators), without having to use single buckets of fuel. If you could either add your own jerry can for liquid fuels or support for portable tanks that would be so sick. Until then I might just have to make an item you craft out of fuel. Would there be a better way of doing this in the meantime, you think?

**Comment by Chris6ix** [comment]
> Having support for fluid tanks would be so sick. I've actually been trying to figure out a way to use fuels from [Create: Diesel Generators](https://www.curseforge.com/minecraft/mc-mods/create-diesel-generators), without having to use single buckets of fuel. If you could either add your own jerry can for liquid fuels or support for portable tanks that would be so sick. Until then I might just have to make an item you craft out of fuel. Would there be a better way of doing this in the meantime, you think?

I agree. This mod is almost perfect. The only thing missing is the ability to use fluids as fuel. I would also like to use gasoline or diesel from the Create: Diesel Generators mod as fuel for ImmersiveAircraft.

**Comment by MCRusher** [comment]
Yup this would be amazing.

Some way to increase fuel capacity would also work in the meantime so that the tank could be filled up with more buckets of gas (etc.) at once and not have to carry 30 buckets to refuel mid flight with.

**Comment by Barerock** [comment]
I would like to request TFMG fuels be added to this list, specifically gasoline, since that's what propeller planes use usually. Additionally, an option to remove compatible fuels, so ONLY gasoline will work, not lava or water.

**Comment by JaoHundred** [comment]
This mod already is the best option in terms of flying vehicles, it only need fluid compatibility. 

lctech tanks as mentioned above would integrate really well with "buy and use" scenarios in an economy driven server, other options could be Mekanism and Oritech tanks. ImmersiveAircraft could also make its own means of fluid fuel like a gas station block maybe.

I hope to see fluid fuels implemented someday.
